### PR TITLE
Assisted teleop simple commander

### PR DESCRIPTION
**Pull request by jwallace42**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | None |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | Gazebo |

---

## Description of contribution in a few bullet points
Add the "assisted teleop" action to the robot navigator api. The contribution also includes a demo for calling the assisted teleop action. 

## Description of documentation updates required from your changes
The updates to the documentation can be found here https://github.com/ros-planning/navigation.ros.org/pull/345.

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_simple_commander/nav2_simple_commander/robot_navigator.py [hunk 1]
```diff
@@ -222,6 +223,26 @@ def backup(self, backup_dist=0.15, backup_speed=0.025, time_allowance=10):
         self.result_future = self.goal_handle.get_result_async()
         return True
 
+    def assistedTeleop(self, time_allowance=0):
```
Comment (SteveMacenski) [inline nav2_simple_commander/nav2_simple_commander/robot_navigator.py:226]:
```
- nav2_simple_commander.test.test_flake8 test_flake8
  <<< failure message
    AssertionError: Found 5 code style errors / warnings:
      ./nav2_simple_commander/example_assisted_teleop.py:16:1: F401 'time.sleep' imported but unused
      ./nav2_simple_commander/example_assisted_teleop.py:18:1: F401 'nav2_simple_commander.robot_navigator.TaskResult' imported but unused
      ./nav2_simple_commander/example_assisted_teleop.py:20:1: F401 'rclpy.duration.Duration' imported but unused
      ./nav2_simple_commander/robot_navigator.py:234:100: E501 line too long (104 > 99 characters)
      ./nav2_simple_commander/robot_navigator.py:246:5: E303 too many blank lines (2)
    assert 1 == 0
  >>>
- nav2_simple_commander.test.test_pep257 test_pep257
  <<< failure message
    AssertionError: Found code style errors / warnings
    assert 1 == 0
  >>>

```

File: nav2_simple_commander/nav2_simple_commander/robot_navigator.py [hunk 2]
```diff
@@ -222,6 +223,26 @@ def backup(self, backup_dist=0.15, backup_speed=0.025, time_allowance=10):
...
+        goal_msg = AssistedTeleop.Goal()
+        goal_msg.time_allowance = Duration(sec=time_allowance)
+
+        self.info("Running 'assisted_teleop'....")
+        send_goal_future = self.assisted_teleop_client.send_goal_async(goal_msg, self._feedbackCallback)
+        rclpy.spin_until_future_complete(self, send_goal_future)
+        self.goal_handle = send_goal_future.result()
+
+        if not self.goal_handle.accepted:
+            self.error('Backup request was rejected!')
```
Comment (SteveMacenski) [inline nav2_simple_commander/nav2_simple_commander/robot_navigator.py:239]: Backup :wink:

File: nav2_simple_commander/nav2_simple_commander/example_assisted_teleop.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    initial_pose.pose.orientation.z = 1.0
+    initial_pose.pose.orientation.w = 0.0
+    navigator.setInitialPose(initial_pose)
+
+    # Wait for navigation to fully activate, since autostarting nav2
+    navigator.waitUntilNav2Active()
+
+    navigator.assistedTeleop(time_allowance=20)
+    while not navigator.isTaskComplete():
+        pass
```
Comment (SteveMacenski) [inline nav2_simple_commander/nav2_simple_commander/example_assisted_teleop.py:47]: Might want to sleep here to throttle the application

Comment (SteveMacenski) [reply nav2_simple_commander/nav2_simple_commander/example_assisted_teleop.py:47]: Add a comment in this while loop about how something like a controller, user command, or other application send command velocities to `xyz` topic during this time for teleoping

File: nav2_simple_commander/launch/assisted_teleop_example_launch.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2022 Joshua Wallace
```
Comment (SteveMacenski) [inline nav2_simple_commander/launch/assisted_teleop_example_launch.py:1]: This is sufficiently derivative you should include the original copyright person for the launch file you copied this from. You can also have your name with 2022 after too.

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
LGTM largely, add to navigation.ros.org docs for migration guide. I've forgotten recently to have people do that since back from vacation :laughing: WIll need to update the migration guide myself at the end of the week for all recent changes

File: nav2_simple_commander/nav2_simple_commander/example_assisted_teleop.py (hunk 3 above)
Comment (SteveMacenski) [reply nav2_simple_commander/nav2_simple_commander/example_assisted_teleop.py:47]: Just so people know that 'oh hey, something else is doing something right now'. Its implied, but we can make it obvious

File: nav2_simple_commander/launch/assisted_teleop_example_launch.py (hunk 4 above)
Comment (jwallace42) [reply nav2_simple_commander/launch/assisted_teleop_example_launch.py:2]: The launch files are pretty much the same except the node being launched. We can probably refactor this at some point.

**Comment by jwallace42** [comment]
CI seems to be stuck looking for a file that got renamed... 

`error: can't copy 'launch/assisted_teleop_demo_launch.py': doesn't exist or not a regular file`

**Comment by SteveMacenski** [comment]
Do the file permissions look like the other python launch files?

**Comment by SteveMacenski** [comment]
Cloned and checked, yes I think the permissions are correct. This is odd. Did you try to create a new file and copy the contents over (or replace the contents of this file with another file we know works, to see if its the content's fault?)

Though, when I clone locally it seems to build fine. Do you see this issue locally?

**Comment by jwallace42** [comment]
I don't see it locally, but it was building correctly before I push up the flake 8 fix(then it failed again). 

It seems to be working now :(.

**Comment by SteveMacenski** [comment]
You have to change all v11's to v12, not just the first one

**Comment by jwallace42** [comment]
> You have to change all v11's to v12, not just the first one

Flipping! Thanks!

File: nav2_simple_commander/package.xml [hunk 5]
```diff
@@ -12,6 +12,7 @@
   <exec_depend>nav2_msgs</exec_depend>
   <exec_depend>action_msgs</exec_depend>
   <exec_depend>lifecycle_msgs</exec_depend>
+  <exec_depend>aws_robomaker_small_warehouse_world</exec_depend>
```
Comment (SteveMacenski) [inline nav2_simple_commander/package.xml:15]: Remove-  I know its actually a dependency, but the package is flaky at being released and I don't want this to block Nav2 binary releases, even if folks need to build themselves locally

File: nav2_simple_commander/nav2_simple_commander/robot_navigator.py (hunk 1 above)
Comment (SteveMacenski) [inline nav2_simple_commander/nav2_simple_commander/robot_navigator.py:226]: What default did we set in the cpp files? I don't think the infinite period is a good default. Maybe 30 seconds?

**Review by SteveMacenski — COMMENTED** [review]
Otherwise LGTM if things are turning over now

**Comment by SteveMacenski** [comment]
Fixed some linting errors in the simple commander that were making CI not green, sorry about that.

File: nav2_simple_commander/nav2_simple_commander/robot_navigator.py (hunk 1 above)
Comment (jwallace42) [reply nav2_simple_commander/nav2_simple_commander/robot_navigator.py]: Switched, there is no default in the cpp file :). So I just set it to 30 sec.

**Review by SteveMacenski — APPROVED** [review]

**Pull request merged**